standalone.go

```diff
@@ -53,29 +64,126 @@ func (s *standalone) pick() int {
 	return rand.IntN(len(s.replicas))
 }
 
+func (s *standalone) handleRedirect(ctx context.Context, cmd Completed, result ValkeyResult) ValkeyResult {
+	if !s.enableRedirect {
+		return result
+	}
+
+	if ret, yes := IsValkeyErr(result.Error()); yes {
+		if addr, ok := ret.IsRedirect(); ok {
+			// Update the latest redirect address before singleflight
+			s.lastRedirectAddr = addr
+			
+			// Use singleflight to ensure only one redirect operation happens at a time
+			// Only proceed if the address matches the latest seen redirect address
+			if err := s.redirectCall.Do(ctx, func() error {
+				// Double-check that this is still the latest redirect address
+				if s.lastRedirectAddr == addr {
+					return s.redirectToPrimary(addr)
+				}
+				// If address doesn't match, another redirect happened, skip this one
```

Comment:
> If address doesn't match, another redirect happened, skip this one

Hmm, I think this is not correct. Here, you are currently holding the singleflight, which means that the one who updated the `lastRedirectAddr` is waiting for you. If you skip the `s.redirectToPrimary(addr)`, then no one will do that.